### PR TITLE
fix: Make sure Xpi class detects and throws an explicit error on invalid chars in zip entries filenames.

### PR DESCRIPTION
**Pull request by rpl**
- Enable yauzl `strictFileNames` option to detect invalid characters (but still require yauzl to send us zip entries filenames and content as strings, to keep that part unchanged)
- Define more explicit Error classes to detect the two error conditions we explicitly expect and want to customize addons-linter errors for without resorting to expect that the error type has to be part of the error message (so that the addons-linter will be able to check the `error.name` property instead)
- Add test coverage for the zip files with an invalid chars (in particular the '\\' used as a separator)

NOTE:
- addons-linter part of this change: https://github.com/mozilla/addons-linter/pull/3940

### CODE REVIEW

File: src/io/xpi.ts [hunk 1]
```diff
@@ -57,7 +58,14 @@ export class Xpi extends IOBase {
 
       this.zipLib.open(
         this.path,
-        { autoClose: this.autoClose },
+        {
+          autoClose: this.autoClose,
+          // Enable checks on invalid chars in zip entries filenames.
+          strictFileNames: true,
+          // Decode automatically filenames and zip entries content from buffer into strings
+          // and autodetects their encoding.
```
Comment (willdurand) [inline src/io/xpi.ts:66]: Maybe add that, without this option set to `true`, `strictFileNames` has no effect.

**Review by willdurand — APPROVED** [review]
r+wc, thanks

File: src/io/xpi.ts (hunk 1 above)
Comment (rpl) [reply src/io/xpi.ts:66]: @willdurand good point, I forgot to make that clear after I did split that into two separate comments

**Comment by willdurand** [comment]
@rpl are we ready to merge this patch and release addons-scanner-utils v5?

**Comment by rpl** [comment]
> @rpl are we ready to merge this patch and release addons-scanner-utils v5?

yes, based on what we discussed last week, this should be ready to merge as is (and the linter PR is the part of this change that needs some more tweaks).

**Comment by willdurand** [comment]
> this should be ready to merge as is

Are you able to merge PRs in this repo?

**Comment by rpl** [comment]
@willdurand nope, it looks that my github permissions associated to this repo do not allow me to merge PRs at the moment.

**Comment by willdurand** [comment]
> @willdurand nope, it looks that my github permissions associated to this repo do not allow me to merge PRs at the moment.

ok, that explains why you didn't merge this PR before I suppose. Sorry for the confusion.

I'll merge this PR myself and cut a new release.

(FWIW, you probably have ["triage" permission](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization) on this repo because we try not to give "write" or more to everyone)

**Pull request merged**